src/main/java/com/iota/iri/service/API.java

```diff
@@ -423,7 +423,7 @@ private AbstractResponse checkConsistencyStatement(List<String> transactionsList
      * @return <tt>false</tt> if we received at least a solid milestone, otherwise <tt>true</tt>
      */
     public boolean invalidSubtangleStatus() {
-        return (snapshotProvider.getLatestSnapshot().getIndex() == snapshotProvider.getInitialSnapshot().getIndex());
+        return latestMilestoneTracker.getLatestMilestoneIndex() -1 > snapshotProvider.getLatestSnapshot().getIndex();
```

Comment:
Change Request:
I think the change is good. I just don't think it should be part of this PR.

Can you revert this and open a new PR to fix #844?
Also can you please change the javadoc of the method?

Thanks

Reply:
To be quite frank, this is super unnecessary from my perspective. Anyway, here you go: https://github.com/iotaledger/iri/pull/1533

Reply:
There are several reasons for asking this:
1. When people (not just me) review a change that is supposed to change a certain behavior they usually want to focus only on the relevant changes to the specified behavior. If people will want to understand how #844 was solved, instead of seeing a small change they will see a big change. 
2. When you go back through history after a few months (or years) to understand when and why something has changed, you want history to be organized.
3. Smaller PRs get reviewed and merged faster in general. 
4. When one does smaller PRs he usually pays more attention to details
5. If a PR changes more than one behavior, and one of them causes a discussion or disagreement, then the other change also gets delayed.

You can maybe argue that in this case some points are less relevant, but in my experience it is best to abide by those rules of thumb because it hard to distinguish when those points will be relevant or not.

Reply:
>When people (not just me) review a change that is supposed to change a certain behavior they usually want to focus only on the relevant changes to the specified behavior. If people will want to understand how #844 was solved, instead of seeing a small change they will see a big change.
I really don't get this. Is this really asking too much?

Please just make sure that both of the PRs are pulled into the same release as there was a reason why it was one unit of fixture for this particular bug (bad syncing).

Reply:
> I really don't get this
I just think that under the umbrella term "syncing issues" you can have lots of changes. It doesn't make it all the same issue.

> Please just make sure that both of the PRs are pulled into the same release
This will happen